src/Standards/Squiz/Sniffs/PHP/NonExecutableCodeSniff.php

```diff
@@ -258,6 +258,16 @@ public function process(File $phpcsFile, $stackPtr)
                 continue;
             }
 
+            // Skip HTML whitespace.
+            if ($tokens[$i]['code'] === T_INLINE_HTML && \trim($tokens[$i]['content']) === '') {
+                continue;
+            }
+
+            // Skip PHP re-open tag (eg, after inline HTML).
+            if ($tokens[$i]['code'] === T_OPEN_TAG) {
```

Comment:
What about if the "re-open tag" is a PHP open tag with echo `<?=` ?

Might be good to add some tests for that as well.

Reply:
Yes, the sniff already handles this case correctly. I've added a test to confirm.

Reply:
I've just tested this, but as far as I can see, the sniff does not handle this correctly, but the test used for this is flawed.

Try this:
```php
<!-- no problem here -->
<?php if (true): ?>
    <?php foreach ([] as $item): ?>
        <?php continue; ?>
        <?= ?>
    <?php endforeach; ?>
<?php endif; ?>

<!-- should detect an error here -->
<?php if (true): ?>
    <?php foreach ([] as $item): ?>
        <?php continue; ?>
        <?=
        'unreachable'
        ?>
    <?php endforeach; ?>
<?php endif; ?>
```

While an empty `<?= ?>` isn't very useful, that's not the concern of the sniff.
If the `<?=` was handled correctly, the first and only error for the above code would be on the line containing the `'unreachable'`. As things are, both lines containing a `<?=` also throw errors.

Reply:
Thanks for the feedback. I've looked into both of these suggested test cases.

The sniff will complain about any _line of code_ which is unreachable. (See https://github.com/squizlabs/PHP_CodeSniffer/blob/d148febc2a2eb82972121d7f962883f7a5697b55/src/Standards/Squiz/Tests/PHP/NonExecutableCodeUnitTest.2.inc#L1-L12 and https://github.com/squizlabs/PHP_CodeSniffer/blob/d148febc2a2eb82972121d7f962883f7a5697b55/src/Standards/Squiz/Tests/PHP/NonExecutableCodeUnitTest.php#L89-L92 - one complaint for each of lines 7-10 from the `exit` on line 3, and an additional complaint for line 10 from the `break` on line 9.)

I didn't see an existing test showing the expected behaviour when there are multiple unreachable statements on one line. I've added a test showing the existing behaviour in 34ba861a8.

Looking at these two suggested sniffs:
- `<?= ?>` is a parse error, so I don't think it needs any special treatment in this sniff. https://3v4l.org/MXtbr
- One statement split across multiple lines also isn't covered by existing tests; I've added a test showing the existing behaviour in 2067e7c6e.
- We could add some logic to detect something like `<?= ''; ?>` which isn't a parse error, and will produce no output, but such a line of code seems like something that should be flagged to developers for review, so I'm comfortable with this sniff reporting that as unreachable. I don't think it's feasible for this sniff to detect output of a variable which happens to be empty and alert differently. A human can review the line of code in that case and make a sensible judgement.

@jrfnl do these cover the concerns raised here, or would you like me to add these specific tests as well?

Reply:
Just to confirm - the tests added in the `2` case file just document existing behaviour. The behaviour regarding these is not changed by this PR. Correct ?

Reply:
Yes, the changes in `src/Standards/Squiz/Tests/PHP/NonExecutableCodeUnitTest.2.inc` (and the corresponding additions in `src/Standards/Squiz/Tests/PHP/NonExecutableCodeUnitTest.php`) are documenting existing behaviour. I can move these to a separate pull request if you would prefer.